From 3f8af498255fb5d198494f5acda09432c662476d Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Fri, 14 Aug 2015 15:11:28 +0800
Subject: [PATCH] MLK-11360-05 ARM: dtsi: imx7d: add snvs clock for rtc, power
 key, and caam nodes

Since snvs-rtc, snvs-pwrkey, snvs-caam all have snvs HP register access
that need snvs clock root enable. Once snvs clock root is defined in i.MX7d
clock driver, it requires device tree to define the clocks property for
snvs related modules like rtc, power key, caam modules.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 arch/arm/boot/dts/imx7d.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index ce08b19..21438c9 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -639,6 +639,8 @@
 			caam_snvs: caam-snvs@30370000 {
 				compatible = "fsl,imx6q-caam-snvs";
 				reg = <0x30370000 0x10000>;
+				clocks = <&clks IMX7D_SNVS_ROOT_CLK>;
+				clock-names = "snvs";
 			};
 
 			snvs: snvs@30370000 {
@@ -652,6 +654,8 @@
 					reg = <0x34 0x58>;
 					interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>,
 						<GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
+					clocks = <&clks IMX7D_SNVS_ROOT_CLK>;
+					clock-names = "snvs";
 				};
 			};
 
@@ -659,6 +663,8 @@
 				compatible = "fsl, imx7d-snvs-pwrkey", "fsl,imx6sx-snvs-pwrkey";
 				reg = <0x30370000 0x10000>;
 				interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clks IMX7D_SNVS_ROOT_CLK>;
+				clock-names = "snvs";
 				fsl,keycode = <116>; /* KEY_POWER */
 				fsl,wakeup;
 			};
-- 
1.8.0

